src/core/state.go

```diff
@@ -648,6 +650,9 @@ func (state *BuildState) WaitForPreloadedSubincludeTargetsAndEnsureDownloaded()
 			wg.Add(1)
 			go func(inc BuildLabel) {
 				state.WaitForTargetAndEnsureDownload(inc, OriginalTarget, true)
+				if err := state.Parser.RegisterPreload(inc); err != nil {
+					panic(err)
```

Comment:
This seems a bit worrying, if the preload fails for any reason (including fairly likely things like errors in the file) this is gonna die with an enormous stacktrace.

Probably we should use an errgroup here to return the errors so they can be handled.

Reply:
Okay I've updated this to handle errors with an errgroup.